uaclient/tests/test_cli_attach.py

```diff
@@ -311,6 +486,93 @@ def test_attach_config_enable_services(
         else:
             assert [] == m_enable.call_args_list
 
+        args = mock.MagicMock(
+            token=None,
+            attach_config=FakeFile(
+                yaml.dump({"token": "faketoken", "enable_services": ["cis"]})
+            ),
+            auto_enable=auto_enable,
+        )
+        args.format = "json"
+
+        fake_stdout = io.StringIO()
+        with contextlib.redirect_stdout(fake_stdout):
+            main_error_handler(action_attach)(args, cfg)
+
+        expected = {
+            "_schema_version": event_logger.JSON_SCHEMA_VERSION,
+            "result": "success",
+            "errors": [],
+            "failed_services": [],
+            "needs_reboot": False,
+            "processed_services": ["cis"] if auto_enable else [],
+            "warnings": [],
+        }
+        assert expected == json.loads(fake_stdout.getvalue())
+
+    @mock.patch("uaclient.contract.process_entitlement_delta")
+    @mock.patch("uaclient.util.apply_series_overrides")
+    @mock.patch("uaclient.contract.UAContractClient.request_url")
+    @mock.patch("uaclient.jobs.update_messaging.update_apt_and_motd_messages")
+    def test_attach_when_one_service_fails_to_enable(
```

Comment:
I'm confused by this test. The name says that one of the services is supposed to fail to enable, but the result doesn't include any failed services

Reply:
Good point, I tried one approach with that test that failed, but I have a better idea now. I have updated the code